lib/cloudflare/connection.rb

```diff
@@ -63,13 +63,17 @@ def paginate(obj, url, url_args = '')
 			# fetch and aggregate all pages
 			loop do
 				query = URI.encode_www_form scope_type: :organization, per_page: page_size, page: page
-				rules = obj.new(concat_urls(url, "?#{query}#{url_args}"), self, **options)
```

Comment:
I found concat_urls was forming URL's with a "/" on the end (blah/?query) which breaks the custom_hostname rest api.

Changing to URI::join in pagnation resolves this and touches as little code as possible.

Reply:
I think that's a good solution.